validate/validate_test.go

```diff
@@ -0,0 +1,1203 @@
+		},
+		{
+			name: "i32.load alignment",
+			code: []byte{
+				// (i32.load align=8 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I32Load, 8, 0,
+			},
+			err: InvalidImmediateError{OpName: "i32.load", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load8s alignment",
+			code: []byte{
+				// (i64.load8_s align=2 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load8s, 2, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load8_s", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load8u alignment",
+			code: []byte{
+				// (i64.load8_u align=2 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load8u, 2, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load8_u", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load16s alignment",
+			code: []byte{
+				// (i64.load16_s align=4 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load16s, 4, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load16_s", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load16u alignment",
+			code: []byte{
+				// (i64.load16_u align=4 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load16u, 4, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load16_u", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load32s alignment",
+			code: []byte{
+				// (i64.load32_s align=8 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load32s, 8, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load32_s", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load32u alignment",
+			code: []byte{
+				// (i64.load32_u align=8 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load32u, 8, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load32_u", ImmType: "naturally aligned"},
+		},
+		{
+			name: "i64.load alignment",
+			code: []byte{
+				// (i64.load align=16 (i32.const 0))
+				operators.I32Const, 0,
+				operators.I64Load, 16, 0,
+			},
+			err: InvalidImmediateError{OpName: "i64.load", ImmType: "naturally aligned"},
+		},
+		{
+			name: "f32.load alignment",
+			code: []byte{
+				// (f32.load align=8 (i32.const 0))
+				operators.I32Const, 0,
+				operators.F32Load, 8, 0,
+			},
+			err: InvalidImmediateError{OpName: "f32.load", ImmType: "naturally aligned"},
+		},
+		{
+			name: "f64.load alignment",
+			code: []byte{
+				// (f64.load align=16 (i32.const 0))
+				operators.I32Const, 0,
+				operators.F64Load, 16, 0,
+			},
+			err: InvalidImmediateError{OpName: "f64.load", ImmType: "naturally aligned"},
+		},
+	}
+
+	for i := range tcs {
+		tc := tcs[i]
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			mod := wasm.Module{}
+			sig := wasm.FunctionSig{Form: 0x60 /* Must always be 0x60 */}
+			fn := wasm.FunctionBody{Module: &mod, Code: tc.code}
+
+			_, err := verifyBody(&sig, &fn, &mod)
+			if err != tc.err {
+				t.Errorf("verify returned '%v', want '%v'", err, tc.err)
```

Comment:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```
(it's in a `testing.T.Run` scope)